src/app/feature-flag/service/feature-acknowledgement.service.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { TestBed } from '@angular/core/testing';
+import { HttpModule, Response, ResponseOptions, XHRBackend } from '@angular/http';
+import { Observable } from 'rxjs';
+
+import { Logger } from 'ngx-base';
+import { WIT_API_URL } from 'ngx-fabric8-wit';
+import { AuthenticationService, UserService } from 'ngx-login-client';
+
+import { loggedInUser } from '../../shared/context.service.mock';
+import { FeatureAcknowledgementService } from './feature-acknowledgement.service';
+
+describe('FeatureAcknowledgement service: it', () => {
+  let extUser = loggedInUser;
+  let featureAckService: FeatureAcknowledgementService;
+  let mockLog: any;
+  let mockAuthService: any;
+  let mockUserService: any;
+
+  beforeEach(() => {
+    mockLog = jasmine.createSpyObj('Logger', ['log']);
+    mockAuthService = jasmine.createSpyObj('AuthenticationService', ['getToken']);
+    mockUserService = jasmine.createSpyObj('UserService', ['getUserByUserId']);
+
+    // Set mock feature acknowledgement flag
+    (extUser.attributes.contextInformation as any).featureAcknowledgement = true;
+
+    mockUserService.getUserByUserId.and.returnValue(Observable.of(extUser));
+    mockUserService.loggedInUser = Observable.of(extUser);
+
+    TestBed.configureTestingModule({
+      imports: [HttpModule],
+      providers: [
+        {
+          provide: Logger, useValue: mockLog
+        },
+        {
+          provide: AuthenticationService,
+          useValue: mockAuthService
+        },
+        {
+          provide: UserService,
+          useValue: mockUserService
+        },
+        {
+          provide: WIT_API_URL,
+          useValue: 'http://example.com'
+        },
+        FeatureAcknowledgementService
+      ]
+    });
+    featureAckService = TestBed.get(FeatureAcknowledgementService);
+  });
+
+  it('should tell whether a user has agree to keep the display of feature icon', () => {
+    featureAckService.getToggle().subscribe(val => {
```

Comment:
This should have a `done()` in it, otherwise this test can pass before `getToggle` emits anything and then spuriously fail later - or simply pass if the Observable never emits. I'd also like to see some way to ensure that the Observable completes, rather than remaining subscribed for the duration of the entire test run (`first()`?)